builder/amazon/common/step_network_info.go

```diff
@@ -0,0 +1,154 @@
+package common
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"math/rand"
+	"sort"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/packer/helper/multistep"
+	"github.com/hashicorp/packer/packer"
+)
+
+// StepNetworkInfo queries AWS for information about
+// VPC's and Subnets that is used throughout the AMI creation process.
+//
+// Produces (adding them to the state bag):
+//   vpc_id string - the VPC ID
+//   subnet_id string - the Subnet ID
+//   availability_zone string - the AZ name
+type StepNetworkInfo struct {
+	VpcId               string
+	VpcFilter           VpcFilterOptions
+	SubnetId            string
+	SubnetFilter        SubnetFilterOptions
+	AvailabilityZone    string
+	SecurityGroupIds    []string
+	SecurityGroupFilter SecurityGroupFilterOptions
+}
+
+type subnetsSort []*ec2.Subnet
+
+func (a subnetsSort) Len() int      { return len(a) }
+func (a subnetsSort) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
+func (a subnetsSort) Less(i, j int) bool {
+	return *a[i].AvailableIpAddressCount < *a[j].AvailableIpAddressCount
+}
+
+// Returns the most recent AMI out of a slice of images.
+func mostFreeSubnet(subnets []*ec2.Subnet) *ec2.Subnet {
+	sortedSubnets := subnets
+	sort.Sort(subnetsSort(sortedSubnets))
```

Comment:
This is not doing exactly what you expect here.
The passed in list will also be sorted

https://play.golang.org/p/wQCtnCBysBP

Reply:
but given the usage, I think this is ok 🙂

Reply:
Yeah this is a bit over-engineered.